sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -202,3 +203,14 @@ object Unions {
     }
   }
 }
+
+/**
+ * Extractor for retrieving Int value.
+ */
+object IntegerIndex {
+  def unapply(a: Any): Option[Int] = a match {
+    case Literal(a: Int, IntegerType) => Some(a)
+    case UnaryMinus(IntegerLiteral(v)) => Some(-v)
```

Comment:
As shown in the following link, our parser convert `-3` to UnaryMinus(IntegerLiteral(3)). Thus, we have to use this to get negative values.

https://github.com/apache/spark/blob/637a78f1d3dff00658324de3887d75c5ccd857be/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/CatalystQl.scala#L641